src/components/widgets/WebWidget.tsx

```diff
@@ -0,0 +1,39 @@
+import { useNavigation } from '@react-navigation/core';
+import React, { useCallback } from 'react';
+
+import { Icon, normalize, texts } from '../../config';
+import { ScreenName, WidgetProps } from '../../types';
+import { Image } from '../Image';
+
+import { DefaultWidget } from './DefaultWidget';
+
+export const WebWidget = ({ text, additionalProps }: WidgetProps) => {
+  const navigation = useNavigation();
+
+  const onPress = useCallback(() => {
+    navigation.navigate(ScreenName.Web, {
+      webUrl: additionalProps?.webUrl,
+      title: additionalProps?.staticContentTitle ?? texts.widgets.web
+    });
+  }, [navigation, text]);
+
+  return (
+    <DefaultWidget
+      Icon={() =>
+        additionalProps?.icon ? (
+          <Image
+            source={{ url: additionalProps?.icon }}
+            style={{
+              height: normalize(additionalProps?.iconSize?.height ?? 26),
+              width: normalize(additionalProps?.iconSize?.width ?? 33)
+            }}
+          />
+        ) : (
+          <Icon.Url size={normalize(26)} />
+        )
```

Comment:
this reads confusing for me. if there is an icon we render an image and otherwise an icon? please clean this up with more appropriate namings.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/932/commits/3e2541dd05a32716c0ef742215a76ae8e67a614c